session-3/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+const AvailableTask int = 100
+
+const (
+	_ = iota
+	Low
+	Medium
+	High
+)
+
+var errNoTasks = errors.New("No tasks completed")
+
+func main() {
+
+	projectName := "Task Management System"
+	startDate := time.Date(2024, 9, 18, 19, 0, 0, 0, time.UTC)
+
+	fmt.Printf("Welcome to the %s\n", projectName)
+	fmt.Printf("Project start date is: %v\n", startDate)
+	fmt.Printf("Project: %s\n", projectName)
+
+	fmt.Println("*****************************")
+
+	completedTask := 75
+	remainingTask := AvailableTask - completedTask
+
+	if completedTask > AvailableTask {
+		fmt.Printf("Error: Completed tasks (%d) exceed the total available tasks (%d) \n", completedTask, AvailableTask)
+		completedTask = AvailableTask
+		remainingTask = 0
+	}
+
+	if completedTask == 0 {
+		err := errNoTasks
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
+		}
+	}
+
+	fmt.Printf("Tasks remaining: %d out of %d\n", remainingTask, AvailableTask)
+
+	isNearCompleted := completedTask >= 90
+	isCompleted := completedTask == AvailableTask
+
+	status := "In Progress"
+
+	if isNearCompleted {
+		fmt.Println("More than 90 tasks completed")
+	}
+
+	if isCompleted {
+		status = "Completed"
+	}
+
+	fmt.Println("Current project status:", status)
+
+	var projectPhase string
+	switch {
+	case completedTask == 0:
+		projectPhase = "Starting phase"
+	case completedTask >= 30 && completedTask < 60:
+		projectPhase = "Midway"
+	case completedTask >= 60:
+		projectPhase = "Final phase"
+	default:
+		projectPhase = "Unknown phase"
+	}
```

Comment:
There is a logical gap versus requirements.
Add a switch statement that categorizes the project progress:
If less than 30 tasks are completed, mark the progress as "Starting phase".
If 30 to 60 tasks are completed, mark it as "Midway".
If more than 60 tasks are completed, mark it as "Final phase".

```suggestion
	switch {
	case completedTask < 30:
		projectPhase = "Starting phase"
	case completedTask >= 30 && completedTask <= 60:
		projectPhase = "Midway"
	case completedTask > 60:
		projectPhase = "Final phase"
	default:
		projectPhase = "Unknown phase"
	}
```

Reply:
yes, it's my issue

Reply:
fixed